### ISSUE TITLE
Home link should have an accessible name

### ISSUE THREAD

**Issue by obulat** [WordPress/openverse-frontend#1714]
## Problem

Currently, the home link has accessible name only on the home page of size `md` and above:
https://github.com/WordPress/openverse-frontend/blob/190d631987449a5417ac80241c53dd9b1ca0d973/src/pages/index.vue#L16-L22

In the header has no accessible name. Instead of saying the home page, Voice Over reads out all of the URL.
<img width="622" alt="Screen Shot 2022-08-20 at 2 42 27 PM" src="https://user-images.githubusercontent.com/15233243/185744432-4754e580-fbae-4667-a427-b390382425f6.png">

Lighthouse shows the following error:
<img width="717" alt="Screen Shot 2022-08-20 at 2 43 48 PM" src="https://user-images.githubusercontent.com/15233243/185744458-8616a611-6078-45cd-9880-921ed04c2963.png">

## Description

According to W3C[^1], we should add an `alt` text to the Openverse logo `<img>` element saying "Openverse home".

## Alternatives

GitHub uses an `aria-label="Homepage"` on the `<a>` link, and makes the image `aria-disabled`.
MDN also uses `aria-label` on the `<a>` element.
Ariakit uses a span with text "Reakit" that's clipped to a size of 1x1px inside of the `<a>` element that also has `aria-current="page"`.

## Additional context

[^1]: https://www.w3.org/WAI/tutorials/images/functional/#example-1-image-used-alone-as-a-linked-logo

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.
